src/VisualStudio/TestUtilities/Workspace/Solution.cs

```diff
@@ -45,13 +45,27 @@ public string FileName
             }
         }
 
+        public void Save()
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(FileName));
+            DteSolution.SaveAs(FileName);
+        }
+
+        public Project OpenProject(string projectFilePath)
```

Comment:
AddExistingProject? Either that or this should be modified with a second parameter that indicates whether the project should be added to the existing solution or added to a new solution.